wordpress-mobile/WordPress-Android
# Add logic to determine if session is subject to consent policies

Fixes #18476

This PR adds logic ported from `woocommerce-android` to identify the country of the active session. There are no UI implementation yet, as they are in scope of another project task, ie #18567.

<details><summary>Some reasoning for why the code isn't an exact copy from woo…</summary>

> I made some small changes to the ported code to make use of Kotlin property delegates, that was only for personal learning and practice, as I felt they might be a clean and easy to understand way of using callable booleans for composing the final result that determines whether users are going to be presented a consent modal or not.
</details>

To test:
1. Apply this diff patch: [`quicktest.patch`](https://github.com/wordpress-mobile/WordPress-Android/files/11707041/quicktest.patch)
2. Build & run `jetpackWasabiDebug` in an emu (emulator)
3. Check `logcat` for debug line with tag `PR_18592_TEST`, the bool value should be true if the device locale country code is in [this list](https://github.com/wordpress-mobile/WordPress-Android/pull/18592/files#diff-767e7314043fe96c916cecff81f30d171280f6c47f8f1a28b68e8513909d9a90R23-R55)
4. Switch emu locale to one of the countries that are opposite to the current one (ie if current locale in list, change to a country which isn't)
6. Restart Jetpack app and check that the value of the log entry is now different

## Regression Notes
1. Potential unintended areas of impact
   N/a

7. What I did to test those areas of impact (or what existing automated tests I relied on)
   N/a

8. What automated tests I added (or what prevented me from doing so)
   N/a - rudimentary logic easy to reason about, no need for tests imho. Perhaps once it's implemented, to make sure we `lock` the behaviour. For now I leave to @mkevins's convenience the freedom to change this code if needed.

PR submission checklist:

- [x] ~~UI Changes testing checklist:~~ n/a
